scripts/build_connectome/connectivity_dmri.py

```diff
@@ -0,0 +1,408 @@
+    """Wrapper function for NiftiMasker to apply mask to image
+
+    Parameters
+    ----------
+    img : str
+        path to input nifti image
+    mask : str
+        path to mask for the nifti image
+    out_masked_img : str
+        name of output masked image
+    """
+    masker = NiftiMasker(mask)
+    masked_img_arr = masker.fit_transform(img)
+    masked_img = masker.inverse_transform(masked_img_arr)
+    masked_img.to_filename(out_masked_img)
+
+
+def atlas2dwi(dwi_b0, mni_nifti, atlas_nifti, atlas_in_dwi_space):
+    """Wrapper function for ANTs to warp atlas from mni space to dwi space
+
+    Parameters
+    ----------
+    dwi_b0 : str
+        path to dwi b0 nifti file
+    mni_nifti : str
+        path to mni nifti template
+    atlas_nifti : str
+        path to atlas nifti file
+    atlas_in_dwi_space : str
+        name of output atlas in dwi space
+    """
+    get_warp_cmd = (
+        f"ANTS 3 -m PR'[{mni_nifti},{dwi_b0}, 1, 2]'"
+        f" -o ANTS -r Gauss'[2,0]' -t SyN'[0.5]' -i 30x99x11"
+        f" --use-Histogram-Matching"
+    )
+
+    apply_warp_cmd = (
+        f"WarpImageMultiTransform 3 {atlas_nifti}"
+        f" {atlas_in_dwi_space} -R {dwi_b0}"
+        f" -i ANTSAffine.txt ANTSInverseWarp.nii --use-NN"
+    )
+
+    print(get_warp_cmd)
+    os.system(get_warp_cmd)
+
+    print(apply_warp_cmd)
+    os.system(apply_warp_cmd)
+
+
+def tck2connectome(
+    atlas,
+    tck,
+    connectivity_matrix,
+    inverse_connectivity_matrix,
+    sift_weights=None,
+):
+    """Wrapper function for MRtrix tck2connectome to create connectivity matrix
+    from tractography
+
+    Parameters
+    ----------
+    atlas : str
+        path to atlas nifti file
+    tck : str
+        path to input .tck file
+    connectivity_matrix : str
+        name of output connectivity matrix (.csv)
+    inverse_connectivity_matrix : str
+        name of output inverse connectivity matrix (.csv)
+    sift_weights : str, optional
+        path to sift weights in .txt format, by default None
+    """
+    if sift_weights is None:
+        # if no sift weights are given, use tck2connectome without
+        # -tck_weights_in parameter
+        # -scale_invnodevol scales the connectivity matrix by inverse of the two
+        #  ROI volumes involved in the connection
+        cmd = (
+            f"tck2connectome -force -symmetric -zero_diagonal "
+            f"-scale_invnodevol {tck} {atlas} {connectivity_matrix} "
+            f"-out_assignments {inverse_connectivity_matrix}"
+        )
+    else:
+        # if sift weights are given, use tck2connectome with
+        # -tck_weights_in parameter
+        cmd = (
+            f"tck2connectome -force -symmetric -zero_diagonal "
+            f"-scale_invnodevol -tck_weights_in {sift_weights} "
+            f"{tck} {atlas} {connectivity_matrix} -out_assignments "
+            f"{inverse_connectivity_matrix}"
+        )
+    print(cmd)
+    os.system(cmd)
+
+
+def pipeline(sub, ses, DATA_ROOT):
```

Comment:
I'd rather have a modular structure with a series of function called by the pipeline.